src/components/CarouselCard/index.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-script-url,max-len */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from './context';
+import './styles.css';
+import Indicators from './indicators';
+
+export default class CarouselCard extends Component {
+    constructor(props) {
+        super(props);
+        this.registerChild = this.registerChild.bind(this);
+        this.setActiveItem = this.setActiveItem.bind(this);
+        this.state = {
+            childs: [],
+            activeItem: undefined,
+            privateRegisterChild: this.registerChild,
+        };
+        this.cardPosition = { transform: 'translateX(-0%)' };
+    }
+
+    componentDidMount() {
+        this.startAnimation();
+    }
+
+    setActiveItem(id) {
+        const { childs } = this.state;
+        const selectedItemIndex = childs.findIndex(child => child.indicatorID === id);
+        this.cardPosition = { transform: `translateX(-${selectedItemIndex}00%)` };
+        this.setState({ activeItem: id });
+    }
+
+    startAnimation() {
+        const { scrollDuration, disableAutoScroll } = this.props;
+        if (!disableAutoScroll) {
+            return setTimeout(() => {
+                const { childs, activeItem } = this.state;
+                const selectedItemIndex = childs.findIndex(child => child.indicatorID === activeItem);
+                const nextItem = selectedItemIndex === childs.length - 1 ? 0 : selectedItemIndex + 1;
+                this.cardPosition = { transform: `translateX(-${nextItem}00%)` };
+                this.startAnimation();
+                this.setState({ activeItem: childs[nextItem].indicatorID });
+            }, scrollDuration);
+        }
+        return undefined;
+    }
+
+    registerChild(child) {
+        const { childs } = this.state;
+        const newChilds = childs.concat([child]);
+        this.setState({ childs: newChilds, activeItem: newChilds[0].indicatorID });
+    }
+
+    render() {
+        const { children } = this.props;
+        const { childs, activeItem } = this.state;
+        return (
+            <div className="rainbow-carousel">
+                <div className="rainbow-carousel__stage">
```

Comment:
I think you can remove this container and put the needed styles in the upper. If it is necessary then change the className because in the convention the ___ character is never used, instead would be _ a single underscore.